java/app/grapheneos/setupwizard/action/WelcomeActions.kt

```diff
@@ -89,4 +95,82 @@ object WelcomeActions {
         if (localeInfo != null) adapter.insert(localeInfo, 0)
         return adapter
     }
+
+    private fun disableOemUnlockByUser() {
+        getOemLockManager()?.isOemUnlockAllowedByUser = false
+        refreshOemUnlockStatus()
+    }
+
+    private fun getOemLockManager(): OemLockManager? {
+        return appContext.getSystemService(OemLockManager::class.java)
+    }
+
+    private fun rebootBootloader() {
+        appContext.getSystemService(PowerManager::class.java)!!.reboot(REBOOT_REASON_BOOTLOADER)
+    }
+
+    fun next(activity: Activity) {
+        if (Build.isDebuggable()) {
+            // we allow free pass for development features on debug builds of the OS
+            SetupWizard.next(activity)
+            return
+        }
+        if (SetupWizard.isSecondaryUser) {
+            // secondary users should not be bothered for this as they're not admins
```

Comment:
Secondary users can be admins too, depending on OS configuration. In any case, I agree that there's no need to show these warnings for them.

Reply:
updated comment